### PR TITLE
Add build-static-release-java.sh to generate Java platform gem

### PR DESCRIPTION
**Pull request by fzakaria**
- Revert a497376abbda332a1b8009c2280157c537f24d56 

- Added a wait step before the build-static-release-java pipeline step.

### CODE REVIEW

File: .buildkite/pipeline.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ steps:
 
   - label: ":mac: build-static-release.sh (master only)"
     command: .buildkite/build-static-release.sh
-    branches: master
+    #branches: master
```
Comment (fzakaria) [inline .buildkite/pipeline.yaml]: @ptarjan I wil have to edit out the commented branches before merging.

Comment (ptarjan) [reply .buildkite/pipeline.yaml]: 👍 feel free to put another commit on this once you see the test run work

**Comment by fzakaria** [comment]
YESS! 

image

File: .buildkite/pipeline.yaml (hunk 1 above)
Comment (fzakaria) [reply .buildkite/pipeline.yaml]: Done!

File: .buildkite/publish-ruby-gems.sh [hunk 2]
```diff
@@ -51,15 +51,25 @@ with_backoff() {
...
-      with_backoff gem push --verbose "$gem_archive"
-  done
+# Push the linux gem
+if ! gem fetch sorbet-static --platform x86_64-linux --version "$release_version" | grep -q "ERROR"; then
+  with_backoff gem push --verbose "_out_/gems/sorbet-static-$release_version-x86_64-linux.gem"
+fi
+
+# Push the mac gem
+if ! gem fetch sorbet-static --platform universal-darwin --version "$release_version" | grep -q "ERROR"; then
+  with_backoff gem push --verbose "_out_/gems/sorbet-static-$release_version-universal-darwin-"*.gem
```
Comment (DarkDimius) [inline .buildkite/publish-ruby-gems.sh]: seems like this broke publishing
https://buildkite.com/sorbet/sorbet/builds/7970#54d3f6d0-8c3d-40f3-9bca-67fb77d24344

Comment (fzakaria) [reply .buildkite/publish-ruby-gems.sh]: Oh cause Darwin has multiple. 
I can fix this asap.

**Pull request merged**